src/VS.Mefx/ConfigCreator.cs

```diff
@@ -0,0 +1,491 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace VS.Mefx
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Globalization;
+    using System.IO;
+    using System.Linq;
+    using System.Reflection;
+    using System.Runtime.Loader;
+    using System.Text.RegularExpressions;
+    using System.Threading.Tasks;
+    using Microsoft.VisualStudio.Composition;
+
+    /// <summary>
+    /// Class to store the catalog and config information for input assemblies.
+    /// </summary>
+    internal class ConfigCreator
+    {
+        /// <summary>
+        /// List of file extensions that are considered valid when trying to find input files.
+        /// </summary>
+        /// <remarks>
+        /// Ensure that the cache extension remains last since the program operates on that assumption.
+        /// </remarks>
+        private static readonly string[] ValidExtensions = { "dll", "exe", "cache" };
+
+        /// <summary>
+        /// List of substrings to look for in that should be ignored.
+        /// </summary>
+        private static readonly string[] InvalidFileStrings = { "System." };
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ConfigCreator"/> class.
+        /// </summary>
+        /// <param name="options">The arguments inputted by the user.</param>
+        public ConfigCreator(CLIOptions options)
+        {
+            this.AssemblyPaths = new List<string>();
+            this.CachePaths = new List<string>();
+            this.PartInformation = new Dictionary<string, ComposablePartDefinition>();
```

Comment:
These first three assignments could be moved to property initializers, and the properties then made read only. For example:

```cs
internal List<string> AssemblyPaths { get; } = new();
```